covquestions-js/src/questionnaireEngine.ts

```diff
@@ -208,25 +208,23 @@ export class QuestionnaireEngine {
           question
         );
         answersFromOptionQuestions.push(processedAnswer);
-        this.data[questionId] = processedAnswer;
+        data[questionId] = processedAnswer;
       } else {
-        this.data[questionId] = rawAnswer;
+        data[questionId] = rawAnswer;
       }
     });
 
-    this.data.score = answersFromOptionQuestions.reduce<ScoreResponse>(
+    data.score = answersFromOptionQuestions.reduce<ScoreResponse>(
       (prev, curr) => this.mergeScores(prev, curr.score),
       {}
     );
-
+    //TODO: Possible bug with variables which reference each other
```

Comment:
Yeah. Currently the variables in the array can only reference variables earlier in the same array. We could do a loop here until the data object does not change anymore (with a limit if someone defines circular dependencies). What do you think?

Reply:
yep, also thought the same way, do you want to fix or create an issue?

Reply:
https://github.com/CovOpen/CovQuestions/issues/127